source/common/upstream/eds.cc

```diff
@@ -72,30 +72,31 @@ void EdsClusterImpl::onConfigUpdate(const ResourceVector& resources) {
   if (updateDynamicHostList(new_hosts, *current_hosts_copy, hosts_added, hosts_removed,
                             health_checker_ != nullptr)) {
     ENVOY_LOG(debug, "EDS hosts changed for cluster: {} ({})", info_->name(), hosts().size());
-    HostListsSharedPtr per_zone(new std::vector<std::vector<HostSharedPtr>>());
+    HostListsSharedPtr per_locality(new std::vector<std::vector<HostSharedPtr>>());
 
-    // If local zone name is not defined then skip populating per zone hosts.
-    if (!local_info_.zoneName().empty()) {
-      std::map<std::string, std::vector<HostSharedPtr>> hosts_per_zone;
+    // If local locality is not defined then skip populating per locality hosts.
+    const Locality local_locality(local_info_.node().locality());
```

Comment:
nit: why copy here? Reference? Or does locality() return a copy?

Reply:
OK I see this is a helper that unpacks the locality.

Reply:
We could do it more efficiently, with `std::string&` in the `std::tuple`, but I think this might be an issue with our multiple `string` implementation today.